### PR TITLE
bugfix(history): history should not to recreate while hot reloading

### PR DESCRIPTION
**Pull request by kuitos**
##### Checklist

- [x] `npm test` passes
- [x] tests are included
- [x] documentation is changed or added
- [x] commit message follows commit guidelines

---
This change is [Reviewable](https://reviewable.io/reviews/umijs/umi/4143)

### CODE REVIEW

File: packages/preset-built-in/src/plugins/generateFiles/umi.tpl [hunk 1]
```diff
@@ -33,6 +33,6 @@ export default clientRender();
 if (module.hot) {
   // @ts-ignore
   module.hot.accept('./core/routes', () => {
-    clientRender();
+    clientRender(true);
```
Comment (sorrycc) [inline packages/preset-built-in/src/plugins/generateFiles/umi.tpl]: 改成 object 形式，方便以后扩展。

```
clientRender({
  hot: true,
})
```

Comment (kuitos) [reply packages/preset-built-in/src/plugins/generateFiles/umi.tpl]: done

File: packages/runtime/src/Plugin/Plugin.test.ts [hunk 2]
```diff
@@ -274,6 +274,17 @@ test('applyPlugins compose', () => {
...
+  ret.length = 0;
+  p.applyPlugins({
+    key: 'foo',
+    type: ApplyPluginsType.compose,
+    initialValue(arg: number) {
+      ret.push(arg);
+    },
+  })(11);
+  expect(applyRet).toEqual('foooo');
+  expect(ret).toEqual([2, 1, 11, 10, 20]);
```
Comment (sorrycc) [inline packages/runtime/src/Plugin/Plugin.test.ts]: 需要同时验证非 initialValue 的 function，参数是否传入正确。

Comment (kuitos) [reply packages/runtime/src/Plugin/Plugin.test.ts]: 非 initialValue 应该是拿不到这个参数的，参数只会一级一级传到 initialValue

**Pull request merged**